core/src/util/yamlUtil.cpp

```diff
@@ -0,0 +1,109 @@
+//
+// Created by Matt Blair on 9/15/18.
+//
+
+#include "yamlUtil.h"
+#include "log.h"
+#include "util/floatFormatter.h"
+#include "csscolorparser.hpp"
+
+namespace Tangram {
+namespace YamlUtil {
+
+glm::vec4 getColorAsVec4(const YAML::Node& node) {
+    double val;
+    if (getDouble(node, val, false)) {
+        return glm::vec4(val, val, val, 1.0);
+    }
+    if (node.IsSequence()) {
+        glm::vec4 vec;
+        if (parseVec(node, vec)) {
+            return vec;
+        }
+    }
+    if (node.IsScalar()) {
+        auto c = CSSColorParser::parse(node.Scalar());
+        return glm::vec4(c.r / 255.f, c.g / 255.f, c.b / 255.f, c.a);
+    }
+    return glm::vec4();
+}
+
+std::string parseSequence(const YAML::Node& node) {
+    if (!node.IsSequence()) {
+        LOGW("Expected a plain sequence: '%'", Dump(node).c_str());
+        return "";
+    }
+
+    std::stringstream sstream;
+    for (const auto& val : node) {
+        if (!val.IsScalar()) {
+            LOGW("Expected a plain sequence: '%'", Dump(node).c_str());
+            return "";
+        }
+        sstream << val.Scalar() << ",";
+    }
+    return sstream.str();
+}
+
+
+bool getInt(const YAML::Node& node, int& result, bool allowTrailingJunk) {
+    if (node.IsScalar()) {
+        auto& str = node.Scalar();
+        char* position = nullptr;
+        constexpr int base = 10;
+        long value = std::strtol(str.data(), &position, base);
```

Comment:
We might have discussed this a while back, but std methods do not handle European number format where `,`->`.` and vice-versa!

Reply:
Or do we not have to worry about this for integers? ..

Reply:
RIGHT! I totally forgot about this. If I can remember now, I think this results in floating point values like `1.234`(a float) being interpreted _in European locales_ as `1234` (an int).

I think rather than struggling to correctly distinguish between floats and ints, it'll be more practical to just parse all numbers as floats or doubles and round if we need to.